rir/src/compiler/analysis/scope.cpp

```diff
@@ -260,6 +260,12 @@ AbstractResult ScopeAnalysis::apply(ScopeAnalysisState& state,
     if (!handled) {
         if (i->hasEnv()) {
             bool envIsNeeded = i->hasEnv();
+
```

Comment:
I am for this change. It is a fix actually. But there is one thing to consider. 
Without this change a deopt was actually not needed. So, we could create stub always in case they are really faster than standard environments. After this change, a deopt is needed in case an instruction creates an environment.

Reply:
good point. do you see a way of checking that?

Reply:
At this point we are actually always deopting -> this change is sound I'd go for it.
What I had in mind was to explore compiling 2 versions and have like an "optimized" deopt that just jump to the version that expects a real environment but does not incurs in many materializations. Was a brainstorming idea but it may be unfeasible or a dead end.